lib/Optimizer/Transforms/LiftArrayAlloc.cpp

```diff
@@ -38,11 +96,12 @@ class AllocaPattern : public OpRewritePattern<cudaq::cc::AllocaOp> {
                                 PatternRewriter &rewriter) const override {
     SmallVector<Operation *> stores;
     bool toGlobal = false;
-    if (!isGoodCandidate(alloc, stores, dom, toGlobal))
+    if (!isGoodCandidate(alloc, stores, dom, toGlobal)) {
       return failure();
+    }
 
     LLVM_DEBUG(llvm::dbgs() << "Candidate was found\n");
-    auto arrTy = cast<cudaq::cc::ArrayType>(alloc.getElementType());
+    auto arrTy = cast<cudaq::cc::ArrayType>(alloc.getType().getElementType());
```

Comment:
This change can be reverted. The AllocaOp class has a getElementType(), which is the call we want to use here.

Reply:
Done!